---
title: "Assignment 1: Data Set Selection and Initial Processing"
output: html_document
---


## Download the Data Set

GEO dataset [GSE224681](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE224681) was chosen for this assignment. First, we should install the R package, "GEOquery", to download dataset from GEO database.
```{r prepares, echo=FALSE}
if (!requireNamespace("BiocManager", quietly = TRUE)) {
    install.packages("BiocManager")
}
if (!requireNamespace("GEOquery", quietly = TRUE)) {
    BiocManager::install("GEOquery")
}

```

Then download the supplement file of GSE224681 dataset:
```{r}
datafile <- "GSE224681_counts.tsv.gz"
if (!file.exists(datafile)) {
    files <- GEOquery::getGEOSuppFiles("GSE224681", makeDirectory = FALSE)
}
data <- read.delim(datafile,header=TRUE,check.names=FALSE,row.names = 1)
# data <- read.table(datafile, header = T, row.names = 1, sep= "\t")
dim(data)
```

```{r}
data[1:10,1:4]
```

The dataset consists of 63925 genes and 6 samples. Each column of the dataset represents one sample. This experimental samples is divided into two groups, one group of 3 replicate samples, with and without treatment by cisplatin.

## Initial Assessment

First, we need to rename the samples:
```{r}
new_data <- data
# rename data colnames
colnames(new_data) <- c(paste0("noCisplatin.", 1:3), paste0("cisplatin.", 1:3))
new_data[1:10,]
```

```{r}
rownames(new_data)[1:10]
```

By plotting the data for all cisplatin-treated and non-cisplatin-treated samples, we understand the distribution and quality of our data and calculate some summary statistics:
```{r}
dataToGraph <- new_data
# add a pseudocount so log2 evaluates
boxplot(log2(dataToGraph + 0.0001), las = 2, ylab="log2 counts", main = "Expression distribution, pre-filtering", cex.axis=0.7)
```

```{r}
summary(log2(dataToGraph + 0.0001))
```

```{r}
length(unique(rownames(new_data))) == nrow(new_data)
```
The result of the last line of code is TRUE, which indicates that all IDs listed as rownames in the dataset are unique.This demonstrates that there is no duplicate gene expression values in the data. 


We divide the samples into two groups, one group of 3 replicate samples:
```{r}
# 2 groups
# 1. noCisplatin
# 2. cisplatin
samples <- data.frame(Treatment = c(rep("noCisplatin",3), rep("cisplatin",3)))
rownames(samples) <- colnames(new_data)
samples
```

## Cleaning the Data

Lowly expressed genes should be removed from both biological and statistical perspectives; biologically, a gene must have a certain amount of expression to be able to transcribe and translate proteins and have biological significance; statistically, an overexpressed gene is not able to obtain significant differences and is not statistically significant.

The first principle to follow is that the count number of a gene should be at least 10-15 in some samples. we can artificially specify that only those above 10 are counted. However, it is slightly better to use cpm instead of raw counts. CPM is well known to eliminate sequencing depth differences.
```{r, echo= FALSE}
if (!requireNamespace("edgeR", quietly = TRUE)) {
    BiocManager::install("edgeR")
}
if (!requireNamespace("limma", quietly = TRUE)) {
    BiocManager::install("limma")
}
```

In this experiment, the number of replicate samples for each group is 3, so we set the threshold of CPM to 3 to filter the rows:
```{r}
cpm_data <- edgeR::cpm(new_data)
rownames(cpm_data) <- rownames(new_data)

filtered_data <- new_data[rowSums(cpm_data >= 1) >= 3, ]

dim(filtered_data)
```


```{r}
length(data[,1]) - length(filtered_data[,1])
```
Using cpm >= 3 for filtering, we finally filtered out 50690 lowly expressed genes and got  13235 features.

## Data Normalization

During sample preparation or sequencing, external factors that do not have biological significance can affect the expression of individual samples. For example, samples prepared in the first batch of an experiment will have higher expression overall than samples prepared in the second batch. Assuming that the range and distribution of expression values should be similar for all samples, normalization is needed to ensure that the expression distribution is similar for each sample throughout the experiment.

There are 7 common normalization methods:

1. Total countï¼ŒTC

2. Upper Quartile, UQ

3. Median, Med

4. DESeq

5. Trimmed Mean of M-values, TMM

6. Quantile

7. Reads Per Kilobase Per Million mapped reads, RPKM



All these methods can be further divided into two categories based on their principles, the first one based on the size of the library (TMM, DESeq) and the second one based on the distribution of the number of reads (TC, UQ, Med, Q, RPKM).

The prerequisite assumption of TMM, DESeq in the first category is that the expression of most genes is not different, and then, based on this assumption a normalization factor is proposed for the criteria based on the mean, or median, proportion, etc. In the case of TMM, however, the ratio is calculated by comparing each sequencing data with the reference data, whereas DESeq considers all samples.

The second type of normalization method presupposes that the distribution of the number of reads should be similar, either in single quartiles (TC, Med, UQ, RPKM), or in all quartiles (Q).

View the distribution of filtered data by plotting a boxplot for all samples:
```{r}
boxplot(log2(edgeR::cpm(filtered_data) + 0.0001), ylab = "log2 CPM", main = "Expression distribution, pre-normalization", cex = 0.5, cex.axis=0.7, las=2)
```

The boxplot shows 6 sets of outliers (all samples). These are 0 counts that I recovered by adding the pseudocount of 0.0001 to my data so log2 doesn't throw an error warning. Let's see how many of these outliers are there:
```{r}
idx <- 1:6
zeroes <- vector(length=6)

# Count number of 0s in each column.
# aka determine number of 0-counts for all samples.
for (i in idx) {
    zero_idx <- which(filtered_data[, i] == 0)
    zeroes[i] <- length(filtered_data[zero_idx,i])
}
zeroes
sum(zeroes)
```

Each element of this vector is the number of zero counts for each samples. We see that they total to 137 outliers. Setting the limits to exclude these outliers:
```{r}
boxplot(log2(edgeR::cpm(filtered_data) + 0.0001), ylab = "log2 CPM", main = "Expression distribution, pre-normalization", cex = 0.5, cex.axis=0.7, las=2, ylim=c(-5,15))
```

In our work flow, we use TMM of edgeR to normalize the data:
```{r}
filtered_matrix <- as.matrix(filtered_data)

# Normalize with respect to treatment type
d <- edgeR::DGEList(counts=filtered_matrix, group=samples$Treatment)
d <- edgeR::calcNormFactors(d)

norm_data <- edgeR::cpm(d)

boxplot(log2(norm_data + 0.0001), ylab = "log2 CPM", main= "Expression distribution, post-normalization", cex = 0.5, cex.axis=0.7, las=2, ylim=c(-5,15))
```

One of the most important exploratory graphs used to examine gene expression analysis is the MDS graph or other similar graphs. These graphs use unsupervised clustering to show similarities and dissimilarities between samples, giving us a general idea of how many differentially expressed genes can be detected before proceeding to a formal test. Ideally, samples would cluster well within different experimental groups, and it would be possible to identify samples far from their groups and track the source of error or additional variance. If there are technical replicates, they should be very close to each other.

Such a plot can be drawn using the plotMDS function in limma. The first plot is before normalization, and the second plot is after normalization.

```{r}
limma::plotMDS(filtered_data, labels=rownames(samples), col=c("red","blue")[factor(samples$Treatment)], main="Sample feature clustering, pre-normalization")
```

```{r}
limma::plotMDS(d, labels=rownames(samples), col=c("red","blue")[factor(samples$Treatment)], main="Sample feature clustering, post-normalization")
```

We see much clearer separation between different groups (cisplatin vs noCisplatin), as well as tighter clustering between replicates of the same treatment group. This is a good outcome and means our normalization was effective.

## Mapping to HUGO Symbols

Supplementary file reveals that the rownames in the data are Ensemble Gene IDs. We must map from Ensemble IDs to HUGO symbols. Same as other gene IDs, Ensemble gene IDs may not match the gene information we want to obtain one-to-one. Some Ensemble gene IDs may not have corresponding symbol IDs, some Ensemble gene IDs may correspond to multiple symbol IDs, and some Ensemble gene IDs have multiple correspondences to one. We should remove the Ensemble gene IDs which are not able to be mapped to the symbol IDs.In the case of multiple symbol IDs for one Ensemble gene ID, we only need to select one of the matching relationships. In the case of multiple Ensemble gene IDs corresponding to one symbol ID, we can merge them by taking the mean, maximum or sum. 

In  our workflow, we merge multiple Ensemble gene IDs corresponding to one symbol ID by taking the sum.We use the bitr function of clusterProfiler to transform the IDs.
```{r,message=FALSE}
if (!requireNamespace("clusterProfiler", quietly = TRUE))
    BiocManager::install("clusterProfiler")

if (!requireNamespace("org.Hs.eg.db", quietly = TRUE))
    BiocManager::install("org.Hs.eg.db")

if (!requireNamespace("stringr", quietly = TRUE)) {
    install.packages("stringr")
}
library(org.Hs.eg.db)
```

```{r}
# Remove the version number of Ensemble Gene ID
gene.ens.id <- rownames(filtered_data)
gene.ens.id <- stringr::str_replace(gene.ens.id, "\\..*","")
rownames(filtered_data) <- gene.ens.id

# Map from Entrez Gene IDs to HUGO symbols
gene.symbols <- clusterProfiler::bitr(geneID = gene.ens.id, 
                    fromType = "ENSEMBL",
                    toType = "SYMBOL",
                    OrgDb = org.Hs.eg.db)
gene.symbols = gene.symbols[!duplicated(gene.symbols$ENSEMBL),]
gene.symbols[1:10,]
```

```{r}
# Remove all unmapped rows
mapped_indices <- rownames(filtered_data) %in% gene.symbols$ENSEMBL
sum(!mapped_indices)
mapped_data <- filtered_data[mapped_indices, ]

# Duplicated HUGO symbols
table(duplicated(gene.symbols$SYMBOL))

# Merge the data based on the same HUGO symbols
# Set HUGO symbols as rownames of new dataframe
mapped_symbols <- gene.symbols$SYMBOL[match(rownames(mapped_data), gene.symbols$ENSEMBL)]
mapped_data <- aggregate(mapped_data, by=list(mapped_symbols), FUN=sum)
mapped_data <- tibble::column_to_rownames(mapped_data,'Group.1')

# Are all HUGO symbols remaining unique? This must return TRUE
nrow(mapped_data) == length(unique(mapped_symbols))
```

```{r}
# Final gene coverage of dataset
dim(mapped_data)
```

```{r}
# Number of rows deleted during mapping?
nrow(filtered_data) - nrow(mapped_data)
```

mapped_data is the final output dataset of this workflow.

## Interpretation

**What are the control and test conditions of the dataset?**

The experiment consists of treating tumor associated macrophages(TAMs) with cisplatin and comparing the induced expression changes with a control group . In this case, the experimental group would be TAMs that are treated with cisplatin, and the control group are TAMs that received no treatment.

**Why is the dataset of interest to me?**

Tumor resistance to chemotherapy and metastatic relapse account for more than 90% of cancer specific mortality. Tumor associated macrophages (TAMs) can process chemotherapeutic agents and impair their action. Little is known about the direct effects of chemotherapy on TAMs.The study explored the role of cisplatin in TAMs. The dataset is from a project studying genes associated with chemotherapy resistance. I thought it would be interesting to explore the mechanism of chemotherapy resistance in TAMs.

**Were there expression values that were not unique for specific genes? How did you handle them?**

Yes. There are 4 repeated identifiers before the mapping to HUGO symbols in the analysis. We merge multiple Ensemble gene IDs corresponding to one symbol ID by taking the sum.

**Were there expression values that could not be mapped to current HUGO symbols?**

Yes, there were. 1034 expression values were unable to be mapped to HUGO symbols. I removed the unmapped rows.

**How many outliers were removed?**

I noticed 6 outliers in my data. I didn't remove any because I felt that their presence was too consistent across one group to simply be measurement error. However, I did reframe the boxplots to exclude these leftover low counts to better see the rest of the data.

**How did you handle replicates?**

I split my data into 2 groups of 3. One group for each combination of time and treatment, and each group contains 3 replicates. I kept each replicate separate, but made sure that replicates of two different groups are clearly differentiated in their names. Finally, I used n = 3 as the low-count threshold for filtering uninformative features, as per edgeR recommendation.

**What is the final coverage of your dataset?**

The final coverage of my dataset is 12197 features across 6 samples.

Summarizing dimensions of data across entire workflow:

```{r}
summary <- list(initial=dim(data), merged_lanes=dim(new_data), removed_low_counts=dim(filtered_data), mapped=dim(mapped_data))
summary
```

